src/generic/imaglist.cpp

```diff
@@ -106,6 +106,10 @@ wxBitmap wxGenericImageList::GetImageListBitmap(const wxBitmap& bitmap) const
 #endif // wxUSE_IMAGE
     }
 
+#ifdef __WXQT__
+    bmpResized.QtBlendMaskWithAlpha();
+#endif
+
```

Comment:
Just trying to understand: is this just an optimization, i.e. would it still work correctly but just slower because it would have to convert the mask multiple times?

Reply:
> Just trying to understand: is this just an optimization, i.e. would it still work correctly...

No it wouldn't. because we have to apply the mask explicitly in:
1- The `wxBitmap` itself. but again, `wxQt` would diverge from the documented behaviour. i.e. converting `wxBitmap` to `wxImage` wouldn't preserve mask information anymore.
2- `wxTreeCtrl` and `wxListCtrl` classes. and I think you agree that it's not wise to call `setMask()` over and over again each time the icons need to be drawn even if the performance hit won't be noticeable in practice.

So I chose to put the call here (in `wxGenericImageList::GetImageListBitmap()`) as is done in wxMSW port by calling MSWBlendMaskWithAlpha() in its own implementation of wxImageList.

> but just slower because it would have to convert the mask multiple times?

Yes see above.

Reply:
OK, the part I didn't realize is that wxQt has native `wx{Tree,List}Ctrl` implementations and so we need to return the native bitmap with mask applied to it when using them. I.e. this wouldn't be necessary if we used generic implementations of these controls, because `wxDC::DrawBitmap()` would draw the bitmap correctly, applying its mask on the fly, but it is needed to work with the native implementations.

If so, i.e. if my understanding is finally correct, then I agree that this is fine. It would, admittedly, be nicer if we added a Qt-specific `wxImageList` implementation too that would inherit from the generic one and just add this `QtBlendMaskWithAlpha()` call, but it would be a lot of work for not that much gain so let's leave it like this.

I'd just like to add a comment like this:
```cpp
// In wxQt the returned bitmap is used with native list/tree controls which require
// the bitmap to already have its mask applied to it, as it's used directly by Qt
// and not via our wxDC::DrawBitmap() which would apply the mask itself, so we must
// take it into account here -- the only alternative would be for the controls to do
// it themselves, but then this would happen every time the bitmap is drawn, which
// would be less efficient than doing it just once here.
```

Again, this is assuming I do understand things correctly now -- if not, please correct me. Otherwise I'll add this comment and merge this soon.

Thanks again!

Reply:
Yes, exactly what you just described and the comment is indeed relevant. and I would be grateful if you add this comment when you merge this. TIA!